File: src/test/java/eotree/TestEOCopy.java
package eotree;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.polystat.j2eo.util.ListUtils.listOf;

import org.junit.jupiter.api.Test;
import org.polystat.j2eo.eotree.EOCopy;
import org.polystat.j2eo.eotree.EODot;

/**
 * EO copying feature tests.
 */
public class TestEOCopy {

    @Test
    public void testGenerateEOZeroIndent() {
        var f = new EOCopy(
                new EODot("name"),
                listOf(
                        new EODot("arg1"),
                        new EODot("arg2")
                        )
        );
        assertEquals("name\n  arg1\n  arg2", f.generateEO(0));
    }

    @Test
    public void testGenerateEONonZeroIndent() {
        var f = new EOCopy(
                new EODot("name"),
                listOf(
                        new EODot("arg1"),
                        new EODot("arg2")
                )
        );
        assertEquals("  name\n    arg1\n    arg2", f.generateEO(1));
    }
}

File: src/test/kotlin/eotree/TestEOCopy.kt
package eotree

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.polystat.j2eo.eotree.EOCopy
import org.polystat.j2eo.eotree.EODot
import org.polystat.j2eo.util.ListUtils.listOf

/**
 * EO copying feature tests.
 */
class TestEOCopy {
    @Test
    fun testGenerateEOZeroIndent() {
        val f = EOCopy(
            EODot("name"),
            listOf(
                EODot("arg1"),
                EODot("arg2")
            )
        )
        Assertions.assertEquals("name\n  arg1\n  arg2", f.generateEO(0))
    }

    @Test
    fun testGenerateEONonZeroIndent() {
        val f = EOCopy(
            EODot("name"),
            listOf(
                EODot("arg1"),
                EODot("arg2")
            )
        )
        Assertions.assertEquals("  name\n    arg1\n    arg2", f.generateEO(1))
    }
}
